File: 백준/Bronze/2884. 알람 시계/알람 시계.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath).toString().split(" ");

solution(parseInt(input[0]), parseInt(input[1]));

function solution(hour, min) {
  if (min - 45 < 0) {
    let minusMin = 60 - 45 + min;
    let minusHour = 0;
    if (hour > 0) {
      minusHour = hour - 1;
    } else {
      minusHour = 24 - 1;
    }
    console.log(`${minusHour} ${minusMin}`);
  } else {
    let minusMin = min - 45;
    console.log(`${hour} ${minusMin}`);
  }
}

File: 백준/Bronze/2884. 알람 시계/알람 시계.py
H,M = map(int,input().split())

if H == 0 :
  H = 24

if M > 45 :
  hour = H
  minute = M - 45
elif M == 45 :
  hour = H
  minute = 0
else :
  minute = M + 15
  if minute == 60 :
    hour = H
    minute = minute
  else :
    hour = H - 1
    minute = minute
if hour > 24 :
  hour = hour - 24
elif hour == 24 :
  hour = 0
print(str(hour) + " " + str(minute))